lib/spack/spack/util/url.py

```diff
@@ -67,11 +67,19 @@ def local_file_path(url):
 def parse(url, scheme="file"):
     """Parse a url.
 
-    For file:// URLs, the netloc and path components are concatenated and
-    passed through spack.util.path.canoncalize_path().
+    Path variable substitution is performed on file URLs as needed.
 
-    Otherwise, the returned value is the same as urllib's urlparse() with
-    allow_fragments=False.
+    Arguments:
+        url (str): URL to be parsed
+        scheme (str): associated URL scheme
+    Returns:
+        (str): For file scheme URLs, the netloc and path components are
+        concatenated and passed through spack.util.path.canoncalize_path().
+        Otherwise, the returned value is the same as urllib's urlparse() with
+        allow_fragments=False.
+
+        (Technically returns ParseResult but cannot get docstring to recognize
+        the associated class.)
```

Comment:
Yes.

Reply:
Does

```
Returns:
    (urllib_parse.ParseResult): ...
```

work? (or e.g. saying `from urllib_parse import ParseResult` at the top). If not, can you post the error here (or if you try it and it fails then I can look at the output for the CI doc check. I'd like to avoid this complication in the docstring if possible.

Reply:
FWIW I had a similar problem this morning and the qualified reference appeared to fix it: https://github.com/spack/spack/pull/31995/commits/a8c28aadc3a6eae036a92bec1af295793dd26544 (for reference)

Reply:
I've had and fixed the `Spec` one before.  I'll try `url lib_parse` (again).

Reply:
Confirmed that does not work as the following warning causes the readthedocs check to fail:

```
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/29026/lib/spack/docs/_spack_root/lib/spack/spack/util/url.py:docstring of spack.util.url.parse:1: WARNING: py:class reference target not found: urllib_parse.ParseResult
```

Reply:
Using `from-import` doesn't help either:

```
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/29026/lib/spack/docs/_spack_root/lib/spack/spack/util/url.py:docstring of spack.util.url.parse:1: WARNING: py:class reference target not found: ParseResult
```

Reply:
Using `import` (vs. `import-as`) doesn't help either:

```
/home/docs/checkouts/readthedocs.org/user_builds/spack/checkouts/29026/lib/spack/docs/_spack_root/lib/spack/spack/util/url.py:docstring of spack.util.url.parse:1: WARNING: py:class reference target not found: six.moves.urllib.parse.ParseResult
```

Reply:
Since it appears to be a problematic class, I added it to the `nitpick_ignore` list maintained in `lib/spack/docs/conf.py` (https://www.sphinx-doc.org/en/master/usage/configuration.html).  It is not the first entry in that list so there is precedence.